base/beam.hpp

```diff
@@ -53,7 +53,15 @@ class Beam
     }
 
     if (m_entries.size() == m_capacity)
+    {
+      if (it == std::prev(m_entries.end()))
+      {
+        m_entries.back() = Entry(key, value);
```

Comment:
Why is it needed?
CC @vng

Reply:
> `pop_back` invalidates `it` if `it == std::prev(m_entries.end())`

Compared to what you did in https://github.com/organicmaps/organicmaps/commit/0764876ead28d001d9a642f6f15974d38159d4d5#diff-a871d373cd08c23fae73306f90a71b35c35e85850ba3ad068ef6bfc61e656914R57, this code does not allow `m_entries` to grow beyond `m_capacity`.

Reply:
Did some checker show the error/warning here? Or you found it by yourself?

Actually, I can't say for sure that the previous code was well-formed .. If the iterator points to the last element pop_back invalidates the element, but the iterator remains _good_ for the follow-up insertion (at least I can't imagine some other vector-iterator implementation).

New code is more safe for sure.

Reply:
> Did some checker show the error/warning here?

MSVC STL detected it. IIRC it was `Beam_Smoke` test.

> If the iterator points to the last element pop_back invalidates the element, but the iterator remains good for the follow-up insertion 

From [cppreference](https://en.cppreference.com/w/cpp/container/vector/pop_back):
> Iterators (including the end() iterator) and references to the last element are invalidated.

My understanding is that "invalid" means "invalid even for insertion".